### ISSUE TITLE
New rule: prefer 'this' or 'super' call when method not overridden in subclass

### ISSUE THREAD

**Issue by fedeci** [eslint/eslint#14035]
**Please describe what the rule should do:**
Let's start with the assumption that if a method of a subclass is not overridden `super` and `this` will call the same exact function.
```js
// e.g.
class A {
  foo() {}
}

class B extends A {
  bar() {
    super.foo()
    // is equal to
    this.foo()
  }
}
```
The rule would enforce the use of `this` or `super` across the codebase when the method called has not been overridden in the subclass.

**What new ECMAScript feature does this rule relate to?**
This is related to classes which is definitely not a recent addition to ES, but I think it might be accepted as it is not a radical change so I propose here :)

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
class Foo {
  foo() {}
}

class Bar extends Foo {
  foo() {} // foo has been overridden
  bar() {
    super.foo() // no suggestion
  }
}
```

```js
// Enforce 'this'
class Foo {
  foo() {}
}

class Bar extends Foo {
  bar() {
    this.foo() // no error
    super.foo() // suggest to use 'this'
  }
}
```

```js
// Enforce 'super'
class Foo {
  foo() {}
}

class Bar extends Foo {
  bar() {
    super.foo() // no error
    this.foo() // suggest to use 'super'
  }
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It is useful to keep coding style consistency across the codebase.

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by ljharb** [comment]
These aren't the same thing though - because classes, like everything else, are mutable, and someone could have changed `Bar.prototype.foo` later - or, someone could be subclassing `Bar` itself, in which case `this.foo()` would call the _subclass'_ method while `super.foo()` would call the superclass' method.

**Comment by fedeci** [comment]
I hadn't thought at that🤦🏻‍♂️
